### PR TITLE
Add configuration types.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR introduces types used in the app configuration (config.yml), and a partial redux state type is added.
The new types are not necessarily exhaustive and may not include obsolete config properties still in example-config.yml.
A few component files have been changed just to give an idea of the direction taken.

 Note that calltaker config types are still to be implemented.

## PR Checklist
- [na] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [na] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/components/map/enhanced-stop-marker.tsx [hunk 1]
```diff
@@ -166,12 +163,11 @@ class EnhancedStopMarker extends Component<Props> {
...
 }
 
-const mapStateToProps = (state: any, ownProps: OwnProps) => {
+const mapStateToProps = (state: AppReduxState, ownProps: OwnProps) => {
   const { highlightedStop } = state.otp.ui
 
-  const transitModes = state.otp.config.modes.transitModes
   const modeColors: ModeColors = {}
-  transitModes.forEach((mode: ConfiguredTransitMode) => {
+  state.otp.config.modes.transitModes.forEach((mode) => {
```
Comment (AdrianaCeric) [inline lib/components/map/enhanced-stop-marker.tsx:170]: Do you want to also add the
`const { config: otpConfig } = state.otp` here?

File: lib/components/util/types.ts [hunk 2]
```diff
@@ -95,10 +95,3 @@ export type SetViewedRouteHandler = (route?: ViewedRouteState) => void
 export type SetViewedStopHandler = (payload: { stopId: string } | null) => void
 
 export type SetLocationHandler = (payload: MapLocationActionArg) => void
-
-export interface ConfiguredTransitMode {
```
Comment (AdrianaCeric) [inline lib/components/util/types.ts:99]: Why is this interface being removed?

**Review by AdrianaCeric — APPROVED** [review]
I think this looks good. Grouping the types in a consistent way and in one file is a good idea. I just had a few questions:

File: lib/components/app/app-menu.tsx [hunk 3]
```diff
@@ -89,7 +90,7 @@ class AppMenu extends Component<
 
   _triggerPopup = () => {
     const { popupTarget, setPopupContent } = this.props
-    setPopupContent(popupTarget)
+    if (popupTarget) setPopupContent(popupTarget)
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply lib/components/app/app-menu.tsx:93]: `popupTarget` is optional, so TypeScript forces this truthy check before triggering the popup.

File: lib/components/map/enhanced-stop-marker.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/map/enhanced-stop-marker.tsx:170]: Not needed, in this file, it is used only on this line.

File: lib/components/util/types.ts (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/components/util/types.ts:99]: I moved it to `config-types.ts`, and it is now called `TransitModeConfig`.

File: example-config.yml [hunk 4]
```diff
@@ -349,8 +349,6 @@ itinerary:
   disableMetroSeperatorDot: false
   # Shows the duration of a leg below the leg in the metro itinerary summary
   showLegDurations: false
-  # Allows calorie counts to be hidden
-  displayCalories: true
```
Comment (miles-grant-ibigroup) [inline example-config.yml:353]: Was this behavior removed?

File: lib/components/narrative/connected-trip-details.js [hunk 5]
```diff
@@ -20,10 +20,6 @@ const mapStateToProps = (state) => {
       mediumId: null,
       riderCategoryId: null
     },
-    displayCalories:
-      typeof itinerary?.displayCalories === 'boolean'
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/connected-trip-details.js:24]: Why was this removed?

**Review by miles-grant-ibigroup — APPROVED** [review]
This is some great cleanup, thanks for all your work!

File: example-config.yml (hunk 4 above)
Comment (binh-dam-ibigroup) [reply example-config.yml:353]: Displaying calories was completely removed from OTP-UI, so this doesn't apply anymore.

File: lib/components/narrative/connected-trip-details.js (hunk 5 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/connected-trip-details.js:24]: Displaying calories was completely removed from OTP-UI, so this doesn't apply anymore.
